admin/services/data-access/pupil.data.service.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+const Pupil = require('../../models/pupil')
+const pupilDataService = {}
+
+/**
+ * Find a single pupil by criteria in `options`
+ * @param options
+ * @return {Promise.<{Object}>}
+ */
+pupilDataService.findOne = async function (options) {
+  const p = await Pupil.findOne(options).populate('school').lean().exec()
+  return p
+}
+
+/**
+ * Update a single object with the new fields from `doc`
+ * @param {string} id Object._id
+ * @param {Object} doc Document to pass to mongo
+ * @return {Promise}
+ */
+pupilDataService.update = async function (id, doc) {
+  return new Promise((resolve, reject) => {
+    Pupil.updateOne({_id: id}, doc, (error) => {
```

Comment:
I am surprised this works with a callback for errors.
isn't there a promisified version of this?

Reply:
you could also promisify it with bluebird I suppose and it would be the same result(?)

Reply:
Not that I know of.

You can put `.exec()` on the end to get a Promise, but then it does not wait for the response from the server.  So not the same thing.

I don't see why you expect it not to work.  Care to elaborate?

Reply:
I am expecting it to work don't get me wrong but I was also expecting it to work under a promise just fine and wait for the server.

Reply:
The underlying mongo api is callback-only. I suspect `updateOne`  is a minimal wrapper around it.

Reply:
Mongoose updateOne returns a query:
http://mongoosejs.com/docs/api.html#model_Model.updateOne

Reply:
the official mongoClient is promisified and much cleaner.  But then we lose the schemas...

Reply:
@pris54 - I'm not sure from your comment if there is a better way to handle this or not.  Can you clarify please?